irmaclient/session.go

```diff
@@ -240,21 +259,57 @@ func (client *Client) newQrSession(qr *irma.Qr, handler Handler) SessionDismisse
 // Core session methods
 
 // getSessionInfo retrieves the first message in the IRMA protocol (only in interactive sessions)
+// If needed, it also handles binding.
 func (session *session) getSessionInfo() {
 	defer session.recoverFromPanic()
 
 	session.Handler.StatusUpdate(session.Action, irma.StatusCommunicating)
 
 	// Get the first IRMA protocol message and parse it
-	err := session.transport.Get("", session.request)
+	info := &irma.ClientRequest{
+		Request: session.request, // As request is an interface, it need to be initialized with a specific instance.
+	}
+	// UnmarshalJSON of ClientRequest takes into account legacy protocols, so we do not have to check that here.
+	err := session.transport.Get("", info)
```

Comment:
I changed it, I had to use reflection in the new `Validate()` function for `ClientRequest` to check whether it is empty or not. Because it is an interface it could not be `nil` and I could have checked one field, but then I'm kind of redoing a part of the `Validate()` method of the session request types.

Reply:
Add comment why reflection is needed.